exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -491,10 +506,18 @@ interface ISqlExpressionBuilder {
         )
 
     /** Returns `true` if this expression is null, `false` otherwise. */
-    fun <T> Expression<T>.isNull(): IsNullOp = IsNullOp(this)
+    fun <T> Expression<T>.isNull() = if (this is Column<*> && columnType.isEntityIdentifier()) {
+        (table as IdTable<*>).mapIdOperator(::IsNullOp)
+    } else {
+        IsNullOp(this)
+    }
 
     /** Returns `true` if this expression is not null, `false` otherwise. */
-    fun <T> Expression<T>.isNotNull(): IsNotNullOp = IsNotNullOp(this)
+    fun <T> Expression<T>.isNotNull() = if (this is Column<*> && columnType.isEntityIdentifier()) {
+        (table as IdTable<*>).mapIdOperator(::IsNotNullOp)
+    } else {
+        IsNotNullOp(this)
+    }
```

Comment:
Both `isNull()` and `isNotNull()` are altered because they are potentially called by a branch in the altered `eq()`/`neq()`. I didn't think that `EntityID` columns would ever be nullable, but setters exist for this case (as seen above), so I included this change.